Design Patterns: Observer Pattern

denisdimitrov opened the pull request:
Observer Pattern for selling products. When a product is sold it notifies all of the 3 observers.

EmilGeorgiev commented on src/main/java/observerPattern/Product.java:
```diff
@@ -0,0 +1,34 @@
+ * @author Denis Dimitrov <denis.k.dimitrov@gmail.com>
+ */
+public class Product implements Subject {
+    private ArrayList<Observer> observers = new ArrayList<>();
```
Use List<Observer>. I want to tell me why this is more appropriate

denisdimitrov replied:
List is probably better because you can later change it from ArrayList to LinkedList. I used ArrayList on purpose because I wanted only array to be used.

denisdimitrov commented:
I changed the ArrayList to List and I moved the method notifyObservers() in the sellProduct() that way everytime a product is sold all of the observers are notified.

EmilGeorgiev commented on src/main/java/observerPattern/Subject.java:
```diff
@@ -0,0 +1,10 @@
+/**
+ * @author Denis Dimitrov <denis.k.dimitrov@gmail.com>
+ */
+public interface Subject {
```
Documentation is important for you and for every one which develop and maintains project.

EmilGeorgiev commented on src/main/java/observerPattern/Subject.java:
```diff
@@ -0,0 +1,10 @@
+public interface Subject {
+    void notifyObservers();
+
+    void addObserver(Observer observer);
```
you have duplication  of word ```Observer``` ```addObserver(Observer)```. Rename method to ``add(Observer)```
